### PR TITLE
KUBOS-198  Doc improvements

### PR DESCRIPTION
**Pull request by kyleparrott**
- [ ] kubos
	- [ ] changelog
		- [ ] Update it :P
	- [ ] main.md
		- [ ] Update all product names to be consistent
			- [ ] Kubos SDK
			- [ ] Kubos CLI
			- [ ] KubOS Linux
			- [ ] KubOS RT
		- [ ] Update file names to be consistent
			- [ ] hypens, not underscores
			- [ ] constistent casing
		- [ ] Move 'contributing' to by changelog (not a sub-document)
		- [ ] Add API link for telemetry
		- [ ] should telemetry-aggregator and ipc also be exposed?
		- [ ] "Installing CLI" - [ ] misleading.  it's also installing vagrant.  Should probably still be "Installing SDK"
		- [ ] "Upgrading CLI" -> Upgrading SDK or Upgrading SDK components
	- [ ] contribution-process
		- [ ] Update 'create a jira issue'
			for stories: follow standard 'as a [user]' format
			mention linking to epics
		- [ ] Change 'create personal copy'
			we're making branches against master now
		- [ ] create a pull request
			verify instructions still accurate
			mention tagging people for review
			mention WIP tag
		- [ ] add 'JIRA -> Reviewing' blurb for stories waiting for PR approval
	- [x] cli-installing - [X] Misnomer.  Really, installing sdk
		- [X] Fix 'kubos-sdk'
		- [X] update vagrant box name
		- [X] Typo: "current Directory"
		- [X] "It is strongly recommended" - [ ] need handholding for this
		- [X] Mounting synced folders - [ ] update instructions.  No example line currently exists
		- [X] Missing 'vagrant up' line
		- [X] A link to the upgrading doc might be nice
	- [x] cli-upgrading
		- [x] Kubos -> KubOS
		- [X] Fix naming (kubos-cli -> KubOS CLI)
		- [X] Add 'to check which version of cli, use `kubos version`'
		- [X] Typo: "avaialble"
		- [X] sudo kubos update -> kubos update
		- [X] add blurb about `kubos use`
		- [x] Add section about downgrading each component (or stick in a different doc)
		- [ ] A link to the install doc might be nice
	- [x] first-project
		- [X] fix naming
		- [X] update vagrant box name
		- [X] Repeat of shared folder note: handholding.  Alternatively, remove this section, since it's not really related to building your first test project.
			- [x] also, is this still the recommended practice?  How do you do `kubos init` on the host machine?  b/c you can't do `kubos init` on a folder that already exists...
		- [ ] Add link for example linux project
	- [x] cli-reference
		- [x] Add links at the top to each section
		- [x] An overview section with all of the kubos commands and the quick description at the top could be nice.  basically just copy paste the output of `kubos --help`, or make a pretty table.
		- [x] Possibly re-org.  Most command references are ordered by command, not by function.
			- [ ] If not re-org, then the doc should be renamed.  SDK Process Overview, or something...
		- [x] go into way more detail on all the things.
		- [x] Make sure all kubos commands and all their options are documented here.
		- [X] "Yotta needs" -> "KubOS CLI needs"
		- [X] "kubos build -- [ ] -v" - [ ] what's the '--' for?
		*** Kubos linking description/instructions could probable use some going over.
		- [X] is there a script for linking everything now?
		- [X] 'TODO' do the thing.
	*** A troubleshooting/FAQ doc could be good.
	- [ ] kubos-development.md
		- [ ] Fix naming
		- [ ] Expand intro blurb with overview of development process.  "If you want to make changes to the Kubos code, perhaps for debugging purposes, you'll need to clone the kubos repo and then link the changed modules to your kubos sdk project" - or something to that effect.
		- [ ] Getting started - [ ] "Modifying an existing Kubos module" doesn't make sense for the text that follows
		- [ ] Since it's recommended to use external shared folders, it might be good to give the process for cloning kubos externally
		and then linking it in the Vagrantfile and running 'vagrant reload''
		- [ ] 'Yotta' - [ ] Make hyperlink to yotta docs
		- [ ] Linking in a local module - [ ] make example directory an actual directory.
			- [ ] "Let's say you want to add debug lines to {module}..." - [ ] Use actual module path.
		- [ ] Change /home/kubos/example to /home/kubos/<project_name> to match previous example `kubos init`
		- [ ] Add instructions for how to unlink modules?
	- [ ] MSP430-launchpad-guide
		- [ ] Fix naming
		- [ ] Update kubos doc links to match formatting of other docs.  "docs/[doc]"
		- [ ] Flashing the board - update the USB passthrough description.  Vagrant should be doing it now.
		- [ ] "sudo kubos flash" - is the sudo still required?
	- [ ] STM32F4-discovery-board-guide
		- [ ] Fix naming
		- [ ] Fix doc URLs for consistency
		- [ ] Fix "sudo kubos flash"?
	- [ ] Linux_Overview
		- [ ] change file name. underscores->hyphen.  Add 'kubos'
		- [ ] Linux -> KubOS Linux
		- [ ] Update busybox commands list
	- [ ] Linux_on_iobc
		- [ ] change file name. underscores->hyphen.  Add 'kubos'
		- [ ] Better highlight that this is a doc specifically for the bootloaders and kernel, not user applications.  Should only be needed if the kernel is not pre-loaded onto the board, or if the kernel needs to be manually updated for some reason.
	- [ ]  User_App_on_iOBC
		- [ ] change file name.
		- [ ] change "KubOS Documentation" -> "Kubos Documentation", since the SDK is lower case, we're just referring to the company's documentation.
		- [ ] Fix zModem file selector example formatting
		- [ ] Example - Add command to exit minicom at very end

### CODE REVIEW

**Comment by kyleparrott** [comment]
This PR diverges from the checklist a little now. I have moved the current reference doc into a "cheatsheet" doc and added a more comprehensive SDK reference doc.

This isn't fully complete but should be complete enough for some 👀  on it.

File: docs/sdk-reference.md [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+    },
+    {
+      "name": "some-test-dependency",
+      "version": "1.5.6",
+      "path": "/some/path/on/disk/yotta_modules/some-test-dependency",
+      "errors": [
+        "a description of some error with this module"
+      ]
+    }
+}
```
Comment (CatFreed) [inline docs/sdk-reference.md]: This is highlighted red for some reason...

File: docs/sdk-reference.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
+# Kubos CLI Command Reference
+
+The `kubos` command is always run with a subcommand in order to do something, `kubos` with no subcommand will only display help information.
+
+## Command Overview
+
```
Comment (CatFreed) [inline docs/sdk-reference.md]: Hyperlinks to each of the commands' descriptions would be awesome

File: docs/sdk-reference.md [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+    link                Symlink a module
+    link-target         Symlink a target
+    list                List the dependencies of the current module, or the inherited targets of the current target.
+    outdated            Display information about dependencies which have newer versions available.
+    remove              Remove or unlink a dependency without removing it from module.json.
+    shrinkwrap          Create a yotta-shrinkwrap.json file to freeze dependency versions.
+    start               Launch the compiled program (available for executable modules only). Requires target support for cross-compiling targets.
+    target              Set or display the target device.
+    test                Run the tests for the current module on the current target. Requires target support for cross-compiling targets.
+    version             Display the current active version of the cli and KubOS source repo.
```
Comment (CatFreed) [inline docs/sdk-reference.md]: Kubos

File: docs/sdk-reference.md [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+The config data is produced by merging the json config data defined by the
+application, the current target, and any targets the current target inherits
+from recursively. Values defined by the application will override those defined
+at the same path by targets, and values defined in targets will override values
+defined by targets they inherit from.
+
+The config data displayed is identical to the data that will be available to
+modules when they are built.
+
+See the config system reference for more details.
```
Comment (CatFreed) [inline docs/sdk-reference.md]: Link?

File: docs/sdk-upgrading.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# Upgrading Your Kubos Development Environment
+
+## Upgrading Kubos CLI
+
+Inside of a vagrant box lives the Kubos CLI. Upon new KubOS releases there may be update for the Kubos CLI.
```
Comment (CatFreed) [inline docs/sdk-upgrading.md]: 'Kubos'
update[s]

File: docs/sdk-upgrading.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+First SSH into your kubos-dev box:
+
+       $ cd <The path containing your kubos-dev Vagrant file>
```
Comment (CatFreed) [inline docs/sdk-upgrading.md]: Vagrantfile

File: docs/sdk-upgrading.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
+# Upgrading Your Kubos Development Environment
+
```
Comment (CatFreed) [inline docs/sdk-upgrading.md]: Blurb about how to check if there's a new version of the CLI and/or Kubos source available?

File: docs/sdk-upgrading.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+        $ sudo pip install --upgrade kubos-cli
+
+## Upgrading the KubOS Source Modules
+
+To update the KubOS source modules your project will be built with run the following command:
+
+        $ kubos update
+
+To check which version of cli, use `kubos version`
```
Comment (CatFreed) [inline docs/sdk-upgrading.md]: Capitalize CLI.  Add "CLI [you're using]"

File: docs/sdk-upgrading.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+        $ kubos use <version number>
+
+The `use` command will checkout and replace the existing KubOS source modules.
+
+After running the `use` command, modules from the new version will be linked.
+If the new release holds new modules or has removed existing modules it may be necessary to re-link all modules to get rid of any module conflicts between new and old versions.
+
+To relink all of the kubos source modules simply run:
+
+        $ kubos link --all
```
Comment (CatFreed) [inline docs/sdk-upgrading.md]: Instructions for downgrading?  "If for some reason the latest version of ((our stuff)) doesn't work, you can downgrade using" blah blah blah...

**Comment by kyleparrott** [comment]
@CatFreed Thank you for the awesome feedback on this. I think it should be ready for another pass.

File: docs/main.md [hunk 10]
```diff
@@ -16,24 +16,26 @@ Looking to build an application on KubOS? Check out our [quick start guide](docs
 
  - [Creating your first project](docs/first-project.md)
 
+ - [KubOS-SDK Cheat Sheet](docs/sdk-cheatsheet.md)
```
Comment (CatFreed) [inline docs/main.md]: Kubos SDK

File: docs/sdk-installing.md [hunk 11]
```diff
@@ -46,23 +46,25 @@ To create an instance of the SDK box follow these steps:
...
+This will create a vagrantfile in your current directory. Vagrantfiles are important as they contain the configuration details for specific boxes.
+Vagrantfiles are dependant on your working directory. To interact with this box in the future you will need to `cd` back to this same directory you have initialized the box in.
 
 
 ### Mounting a host directory:
 
 It is strongly recommended that you create your project in a directory on your host that is shared with your box. By keeping your project on your host it will protect them in the event your box is destroyed or re-built.
 
-To mount a specific directory from your host, open the Vagrantfile that was in the kubos-vagrant directory you clone in the above step and look for the following lines:
+To mount a specific directory from your host, open the Vagrantfile located in the directory you ran `vagrant init` insdie of in the above step and look for the following lines:
```
Comment (CatFreed) [inline docs/sdk-installing.md]: nit: spelling 'insdie'

File: docs/sdk-reference.md [hunk 12]
```diff
@@ -4,25 +4,25 @@ The `kubos` command is always run with a subcommand in order to do something, `k
...
-    remove              Remove or unlink a dependency without removing it from module.json.
-    shrinkwrap          Create a yotta-shrinkwrap.json file to freeze dependency versions.
-    start               Launch the compiled program (available for executable modules only). Requires target support for cross-compiling targets.
-    target              Set or display the target device.
-    test                Run the tests for the current module on the current target. Requires target support for cross-compiling targets.
-    version             Display the current active version of the cli and KubOS source repo.
-    versions            Display the available versions of the KubOS source.
-    update              Download newer versions of the KubOS Modules
-    use                 Set a new version of the KubOS modules to build your projects against.
+    [build](#kubos-build)               Build the current module.
```
Comment (CatFreed) [inline docs/sdk-reference.md]: hyperlinks don't work if they're formatted as a code block

File: docs/sdk-upgrading.md [hunk 13]
```diff
@@ -1,27 +1,29 @@
...
 
 To update the KubOS source modules your project will be built with run the following command:
 
         $ kubos update
 
-To check which version of cli, use `kubos version`
+To check which version of the CLI you're using, use `kubos version`
 
-To list all of the available versions of the KubOS source modules run:
+To list all of the available versions of the KubOS s4ource modules run:
```
Comment (CatFreed) [inline docs/sdk-upgrading.md]: 's4ource'

File: docs/main.md [hunk 14]
```diff
@@ -7,7 +7,7 @@ Looking to build an application on KubOS? Check out our [quick start guide](docs
 ## Docs
 
  - [Changelog](docs/changelog.md)
- 
+
 ### KubOS SDK
 
  - [Installing Kubos-CLI](docs/cli-installing.md)
```
Comment (CatFreed) [inline docs/main.md]: Kubos SDK.  doc name -> sdk-installing

File: docs/first-project.md [hunk 15]
```diff
@@ -1,14 +1,14 @@
-# Getting started with KubOS-SDK
+# Getting started with Kubos SDK
 
 This is intended to be a quick guide to creating a new project on top of the Kubos framework.
 
 ## Prerequisites
 
-[Install the KubOS-CLI](docs/cli-installing.md)
+[Install the Kubos CLI](docs/cli-installing.md)
```
Comment (CatFreed) [inline docs/first-project.md]: Kubos SDK.  doc -> sdk-installing

File: docs/sdk-upgrading.md [hunk 16]
```diff
@@ -0,0 +1,45 @@
+# Upgrading Your Kubos Development Environment
+
+New updates will be announced on the Kubos website. There will be instructions depending on the specifc release details of all the steps needed to upgrade for the newest release.
```
Comment (CatFreed) [inline docs/sdk-upgrading.md]: Add link to kubos website. 'specifc'

File: docs/first-project.md [hunk 17]
```diff
@@ -54,6 +60,14 @@ If you would prefer to use one of our other examples as a starting point all you
 
         $ git clone https://github.com/kubos-rt-example myproject
 
+If you created your project from a clone there's some additional setup needed to satisfy all of the projects depedencies for Kubos source modules.
```
Comment (CatFreed) [inline docs/first-project.md]: nit: 'depedencies'

File: docs/main.md [hunk 18]
```diff
@@ -7,33 +7,35 @@ Looking to build an application on KubOS? Check out our [quick start guide](docs
...
 
  - [Changelog](docs/changelog.md)
- 
-### KubOS SDK
 
- - [Installing Kubos-CLI](docs/cli-installing.md)
+### Kubos SDK
 
- - [Upgrading Kubos-CLI](docs/cli-upgrading.md)
+ - [Installing Kubos CLI](docs/cli-installing.md)
```
Comment (CatFreed) [inline docs/main.md]: Kubos SDK.  doc -> sdk-installing

File: docs/main.md [hunk 19]
```diff
@@ -7,33 +7,35 @@ Looking to build an application on KubOS? Check out our [quick start guide](docs
...
+
+ - [Upgrading Kubos CLI](docs/cli-upgrading.md)
```
Comment (CatFreed) [inline docs/main.md]: Kubos SDK. sdk-upgrading

File: docs/main.md [hunk 20]
```diff
@@ -14,9 +14,13 @@ If for some reason Slack won't work for you, feel free to email us at info@kubos
 
 ### Kubos SDK
 
- - [Installing Kubos CLI](docs/cli-installing.md)
+ - [Installing Kubos SDK](docs/sdk-installing.md)
 
- - [Upgrading Kubos CLI](docs/cli-upgrading.md)
+ - [Contributing to the Kubos Project](docs/contribution-process.md)
+
+ - [Kubos Naming and Coding Standards](docs/kubos-standards.md)
```
Comment (CatFreed) [inline docs/main.md]: Why did you move contributing and standards here?  IMO they should still be higher-level docs under the main 'Docs' header.

File: docs/main.md [hunk 21]
```diff
@@ -28,6 +32,11 @@ If for some reason Slack won't work for you, feel free to email us at info@kubos
 
  - [Contributing to the Kubos Project](docs/contribution-process.md)
```
Comment (CatFreed) [inline docs/main.md]: Duplicate

File: docs/main.md [hunk 22]
```diff
@@ -11,34 +11,53 @@ If for some reason Slack won't work for you, feel free to email us at info@kubos
...
  - [Changelog](docs/changelog.md)
- - [Contributing to the Kubos Project](docs/contribution-process.md)
- - [Kubos Naming and Coding Standards](docs/kubos-standards.md)
- 
+
 ### Kubos SDK
 
  - [Installing Kubos SDK](docs/sdk-installing.md)
 
+ - [Contributing to the Kubos Project](docs/contribution-process.md)
```
Comment (kyleparrott) [inline docs/main.md]: Already fixing the order and duplicates of these

File: docs/main.md [hunk 23]
```diff
@@ -28,6 +32,11 @@ If for some reason Slack won't work for you, feel free to email us at info@kubos
 
  - [Contributing to the Kubos Project](docs/contribution-process.md)
 
+ - [Kubos CLI Command Reference](docs/sdk-reference.md)
+
+ - [Kubos Module Development](docs/kubos-development.md)
```
Comment (CatFreed) [inline docs/main.md]: These two are duplicates, too

File: docs/main.md [hunk 24]
```diff
@@ -43,6 +52,15 @@ If for some reason Slack won't work for you, feel free to email us at info@kubos
...
 
+### KubOS Linux
+
+ - [KubOS Linux Overview](docs/kubos-linux-overview.md)
+
+ - [KubOS Linux on iOBC](docs/kubos-linux-on-iobc.md)
+
+ - [User Applications on iOBC](docs/user-app-on-iobc.md)
+
+
```
Comment (CatFreed) [inline docs/main.md]: Duplicate section

File: docs/main.md (hunk 22 above)
Comment (CatFreed) [inline docs/main.md]: Well, it's too late, i already submitted my comments on them :D

**Comment by CatFreed** [comment]
:shipit:

File: docs/sdk-cheatsheet.md [hunk 25]
```diff
@@ -137,12 +151,13 @@ TODO: We probably want to add some info here on Virtualbox device filters and Gu
...
 
 A gdb server must be started to allow your gdb instance to connect and debug directly on your hardware device.
 After building your project with `kubos build` the kubos-cli can start a gdb server and gdb instance for you.
 
 Start a gdb server and instance:
-Note: this may need to run as root depending on your usb device permissions
+Note: this may need to run as root depending on your USB device permissions
 
         $ kubos debug
+
```
Comment (plauche) [inline docs/sdk-cheatsheet.md]: It's probably worth noting that at this point you will be inside of an active gdb instance.
